Source/Core/Duality/Cloning/Surrogates/HashSetSurrogate.cs

```diff
@@ -0,0 +1,67 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Reflection;
+
+namespace Duality.Cloning.Surrogates
+{
+	public class HashSetSurrogate : CloneSurrogate<dynamic>
+	{
+		public override bool MatchesType(TypeInfo t)
+		{
+			return
+				t.IsGenericType &&
+				t.GetGenericTypeDefinition() == typeof(HashSet<>);
+		}
+
+		public override void SetupCloneTargets(dynamic source, dynamic target, ICloneTargetSetup setup)
+		{
+			Type hashSetType = source.GetType();
+			Type[] genArgs = hashSetType.GenericTypeArguments;
+			TypeInfo firstGenArgInfo = genArgs[0].GetTypeInfo();
+
+			if (!firstGenArgInfo.IsPlainOldData())
+			{
+				if (source == target)
+				{
+					foreach (dynamic key in source)
+						setup.HandleObject(key, key);
+				}
+				else
+				{
+					foreach (dynamic key in source)
```

Comment:
See above

Reply:
Yes if you use object you will get a runtimebinding exception as the method expects a T and you give it a object. Using dynamic will actually resolve this properly instead of giving a exception at runtime.